files/ko/web/api/xmlhttprequest/index.md

```diff
@@ -0,0 +1,122 @@
+---
+title: XMLHttpRequest
+slug: Web/API/XMLHttpRequest
+tags:
+  - AJAX
+  - API
+  - Communication
+  - HTTP
+  - Interface
+  - Reference
+  - Web
+  - XHR
+  - XMLHttpRequest
+browser-compat: api.XMLHttpRequest
+translation_of: Web/API/XMLHttpRequest
+---
+
+{{DefaultAPISidebar("XMLHttpRequest")}}
+
+`XMLHttpRequest` (XHR) 객체는 서버와 상호작용할 때 사용합니다. XHR을 사용하면 페이지의 새로고침 없이도 URL에서 데이터를 가져올 수 있습니다. 이를 활용하면 사용자의 작업을 방해하지 않고 페이지의 일부를 업데이트할 수 있습니다.
+
+`XMLHttpRequest`는 [AJAX](/ko/docs/Web/Guide/AJAX) 프로그래밍에 많이 사용됩니다.
+
+{{InheritanceDiagram}}
+
+이름에 `XML`이 들어가긴 하지만, XHR은 XML 뿐만 아니라 모든 종류의 데이터를 가져올 수 있습니다.
+
+이벤트 데이터나 메시지 데이터를 서버에서 가져와야 하는 통신의 경우, {{domxref("EventSource")}} 인터페이스를 통한 [서버발 이벤트](/ko/docs/Web/API/Server-sent_events)(Server-sent events)의 사용을 고려하세요. 완전한 양방향 통신의 경우에는 [WebSocket](/ko/docs/Web/API/WebSockets_API)이 더 좋은 선택일 수 있습니다.
+
+{{AvailableInWorkers("notservice")}}
+
+## 생성자
+
+- {{domxref("XMLHttpRequest.XMLHttpRequest", "XMLHttpRequest()")}}
+  - : `XMLHttpRequst`를 초기화합니다.
+
+## 속성
+
+{{domxref("XMLHttpRequestEventTarget")}}과 {{domxref("EventTarget")}}의 속성을 상속합니다.
+
+- {{domxref("XMLHttpRequest.readyState")}} {{readonlyinline}}
+  - : 요청의 상태를 나타내는 숫자를 반환합니다.
+- {{domxref("XMLHttpRequest.response")}} {{readonlyinline}}
+  - : {{domxref("XMLHttpRequest.responseType")}}의 값에 따라, 응답 개체 본문을 포함하는 {{jsxref("ArrayBuffer")}}, {{domxref("Blob")}}, {{domxref("Document")}}, JavaScript 객체, 또는 {{domxref("DOMString")}}을 반환합니다.
+- {{domxref("XMLHttpRequest.responseText")}} {{readonlyinline}}
+  - : 요청에 대한 응답을 텍스트로 나타내는 {{domxref("DOMString")}}을 반환합니다. 요청이 실패했거나, 아직 전송하지않은 경우에는 `null`을 반환합니다.
+- {{domxref("XMLHttpRequest.responseType")}}
+  - : 응답의 유형을 지정합니다.
+- {{domxref("XMLHttpRequest.responseURL")}} {{readonlyinline}}
+  - : 응답의 URL을 직렬화한 값을 반환합니다. URL이 `null`이면 빈 문자열을 반환합니다.
+- {{domxref("XMLHttpRequest.responseXML")}} {{readonlyinline}}
+  - : 요청에 대한 응답을 포함한 {{domxref("Document")}}를 반환합니다. 요청이 실패했거나, 아직 전송하지 않았거나, 응답을 XML/HTML로 파싱할 수 없는 경우 `null`을 반환합니다. [Web Worker](/ko/docs/Web/API/Web_Workers_API)에서 사용할 수 없습니다.
+- {{domxref("XMLHttpRequest.status")}} {{readonlyinline}}
+  - : 응답의 [HTTP 상태 코드](/ko/docs/Web/HTTP/Status)를 반환합니다.
+- {{domxref("XMLHttpRequest.statusText")}} {{readonlyinline}}
+  - : HTTP 서버가 반환한 응답 문자열을 담은 {{domxref("DOMString")}}을 반환합니다. {{domxref("XMLHttpRequest.status")}}와 달리, `statusText`는 "`OK`"와 같은 응답 상태 메시지의 전문을 반환합니다
+
+    > **참고:** HTTP/1.1의 상태 텍스트와 같이 프로토콜 버전과 응답의 이유를 알릴 방법이 HTTP/2 명세에는 없습니다. ({{RFC(7540, "Response Pseudo-Header Fields", "8.1.2.4")}})
+
+- {{domxref("XMLHttpRequest.timeout")}}
+  - : 요청을 자동으로 종료하기 전에 대기할 시간을 나타내는 밀리초 단위 값입니다.
+- {{domxref("XMLHttpRequest.upload")}} {{readonlyinline}}
+  - : 업로드 과정을 나타내는 {{domxref("XMLHttpRequestUpload")}}입니다.
+- {{domxref("XMLHttpRequest.withCredentials")}}
+  - : 사이트 간 `Access-Control` 요청에 쿠키나 인증 헤더 등 자격 증명을 사용해야 하는지 여부를 나타냅니다.
```

Comment:
```suggestion
  - : 쿠키 또는 권한 부여 헤더와 같은 자격 증명을 사용하여 사이트 간 `Access-Control` 요청을 만들어야 하는 경우 `true`를 반환합니다. 그렇지 않으면 `false`를 반환합니다.
```

> Returns true if cross-site Access-Control requests should be made using credentials such as cookies or authorization headers; otherwise false.